middleware/tls.go

```diff
@@ -0,0 +1,26 @@
+package middleware
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+)
+
+
+func NewTLSConfig(certPath, keyPath, caPath string) (*tls.Config, error) {
```

Comment:
think we use this in `etcd` middleware as well, should probably be moved to middleware/pkg/tls